### PR TITLE
Adds kube-oidc-proxy config

### PR DESCRIPTION
**Pull request by jr0d**
Currently blocked on: https://github.com/mesosphere/charts/pull/143

### CODE REVIEW

File: templates/konvoyconfig.yaml [hunk 1]
```diff
@@ -21,11 +21,15 @@ spec:
...
     repo: https://mesosphere.github.io/charts/staging
-    version: 0.0.1
+    version: 0.0.2
     values: |
       ---
       config:
         # clusterHostname should be a FQDN that resolves to the cluster's ingress VIP.
         # This value will be used to generate certificates for traefik and configure dex.
         clusterHostname: ""
+        # External Dex data for kube-oidc-proxy
```
Comment (shaneutt) [inline templates/konvoyconfig.yaml]: I appreciate example comments like this :+1:

File: templates/kube-oidc-proxy.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+apiVersion: kubeaddons.mesosphere.io/v1alpha1
+kind: Addon
+metadata:
+  name: kube-oidc-proxy
+  namespace: kubeaddons
+  labels:
+    kubeaddons.mesosphere.io/name: kube-oidc-proxy
+  annotations:
+    appversion.kubeaddons.mesosphere.io/dex-k8s-authenticator: "v1.1.0"
+    values.chart.helm.kubeaddons.mesosphere.io/dex-k8s-authenticator: "*"
```
Comment (shaneutt) [inline templates/kube-oidc-proxy.yaml]: This should be a link, we're not trying to keep these annotations around long-term but at least for now let's make sure we add them.

File: templates/kube-oidc-proxy.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      oidc:
+        clientId: kube-apiserver
+        # updated by initcontainer when working with a remote dex
+        issuerUrl: https://dex-kubeaddons.kubeaddons.svc.cluster.local:8080/dex
+        usernameClaim: email
+        # placeholder : updated by initcontainer
+        caPEM: "placeholder"
+      initContainers:
+      - name: initialize-kube-oidc-proxy
+        image: jaredrodriguez/kubeaddons-addon-initializer:latest
```
Comment (s12chung) [inline templates/kube-oidc-proxy.yaml]: let's use the our shared docker hub repo with a static tag, `latest` is a bit dangerous for actual deployments

Comment (hectorj2f) [reply templates/kube-oidc-proxy.yaml]: +1

Comment (hectorj2f) [reply templates/kube-oidc-proxy.yaml]: @s12chung can you tag a new version of kubeaddons-addon-initializer with the latest changes ?

Comment (s12chung) [reply templates/kube-oidc-proxy.yaml]: building...

Comment (jr0d) [reply templates/kube-oidc-proxy.yaml]: Thanks, this was here only for testing.

**Pull request merged**